infrastructure/comms/src/peer_manager/peer.rs

```diff
@@ -20,18 +20,85 @@
 //  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 //  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-use crate::connection::NetAddress;
+use crate::{connection::net_address::net_addresses::NetAddresses, peer_manager::node_id::NodeId};
+use chrono::prelude::*;
+use tari_crypto::keys::PublicKey;
+use tari_utilities::bit::Bits;
 
+// TODO reputation metric?
+
+#[derive(Debug)]
 pub enum PeerType {
     BaseNode,
     ValidatorNode,
     Wallet,
     TokenWallet,
 }
 
-#[allow(dead_code)]
-pub struct Peer {
-    // TODO: Add fields
-    peer_type: PeerType,
-    addresses: Vec<NetAddress>,
+#[derive(Debug)]
+pub struct Peer<K: PublicKey> {
+    pub public_key: K,
+    pub node_id: NodeId,
+    pub peer_type: PeerType,
+    pub addresses: NetAddresses,
+    pub flags: u8,
+}
+
+impl<K> Peer<K>
+where K: PublicKey
+{
+    /// Constructs a new peer
+    pub fn new(public_key: K, node_id: NodeId, peer_type: PeerType, addresses: NetAddresses, flags: u8) -> Peer<K> {
+        Peer {
+            public_key,
+            node_id,
+            peer_type,
+            addresses,
+            flags,
+        }
+    }
+
+    /// Provides that date time of the last successful interaction with the peer
+    pub fn last_seen(&self) -> Option<DateTime<Utc>> {
+        self.addresses.last_seen()
+    }
+
+    /// Returns the ban status of the peer
+    pub fn is_banned(&self) -> bool {
+        self.flags.get_bit(0)
```

Comment:
Rather look at the `bitflags` crate for this; it's a big readablility win

Reply:
Made use of the bitflags crate.